soroban-env-host/src/host.rs

```diff
@@ -634,9 +634,10 @@ impl Host {
 
     pub(crate) fn charge_for_new_host_object(
         &self,
+        handle: usize,
         ho: HostObject,
     ) -> Result<HostObject, HostError> {
-        self.charge_budget(CostType::HostObjAllocSlot, 1)?;
+        self.charge_budget(CostType::HostObjAllocSlot, handle as u64)?;
```

Comment:
I think this is the first place we are passing a handle around outside of a `RawVal` or an `Object`. Given that `Object` is almost free, it might be worth passing the handle around as an `Object` and extracting its handle at this line.

Reply:
It's a synonym for "size of the object array" here -- the handle allocated is always "the previous size of the array". I can rename it for clarity.